packages/amplify-cli-core/API.md

```diff
@@ -182,6 +182,25 @@ export enum AmplifyFrontend {
     javascript = "javascript"
 }
 
+// @public (undocumented)
+export class AmplifyNodePkgDetector {
+    constructor(amplifyDetectorProps: AmplifyNodePkgDetectorProps);
+    // Warning: (ae-forgotten-export) The symbol "DetectedDependencies" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
+    detectAffectedDirectDependencies(): Array<DetectedDependencies> | undefined;
+    // Warning: (ae-forgotten-export) The symbol "Lockfile" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
+    parseLockFile(): Lockfile;
```

Comment:
this shouldn't be public.

Reply:
I have kept this public for 2 reasons:
1) if someone want to use the detector to parse lock files in future
2) helps in testing the detector class for both parsers (though this I can change)

Reply:
Public api (assuming that we treat semver seriously) is an element of a library/package that's hardest to change. Therefore we shouldn't overexpose it if there's no strong case to do that - tests shouldn't be a driver here. Besides using detector to parse doesn't seem like a cohesive component (i.e. it stops being responsible for single thing).

Consider the following:
- the need of using cli core to parse lock files outside of this package is speculative at this time. therefore publishing such interface should be deferred until needed (following YAGNI philosophy).
- if we were to provide lockfile parsing facility publicly, we'd rather publish parsers (most likely in a form of interface and type factory for the interface, but this would need another review and consideration, we don't do this until a need arises).
- parsing looks quite surprising on "detector". the detector might use parser internally through delegation but exposing parse method seems like violation of srp principle here.